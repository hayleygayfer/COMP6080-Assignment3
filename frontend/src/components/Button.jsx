import React from 'react';
import PropTypes from 'prop-types';
function Button (props) {
  return <button className='button' onClick={props.onClickFunction}>{props.text}</button>
}
export default Button;
Button.propTypes = {
  text: PropTypes.string,
  onClickFunction: PropTypes.func
};
