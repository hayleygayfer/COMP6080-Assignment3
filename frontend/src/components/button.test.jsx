import { shallow } from 'enzyme';
import * as React from 'react';
import Button from './Button'
import renderer from 'react-test-renderer';

describe('AddQuestion', () => {
  // const mockCallBack = jest.fn();
  const noop = () => {};

  it('triggers onClick event handler when clicked', () => {
    const onClick = jest.fn();
    //create buttpm. 'click' on it, check if it was clicked
    const button = shallow(<Button text='Login' onClickFunction={onClick}/>);
    button.simulate('click')
    expect(onClick).toBeCalledTimes(1);
  });

  it('uses given text through props', () => {
    const customText = 'custom text';
    const button = shallow(<Button text={customText} onClickFunction={noop}/>);
    expect(button.text()).toBe(customText) //check if text matches
  })

  // SNAPSHOT TESTING....
  it('uses given text through props', () => {
    const customText = 'custom text';
    const button = renderer.create(<Button text={customText} onClickFunction={noop}/>).toJSON();
    expect(button).toMatchSnapshot();
  })

});
