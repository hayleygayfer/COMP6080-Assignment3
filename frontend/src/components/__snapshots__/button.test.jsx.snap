// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddQuestion uses given text through props 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  custom text
</button>
`;
